pkg/auth/manager.go

```diff
@@ -128,10 +130,15 @@ func handleAuthentication(a *authManager, k authKey, reAuth bool) {
 			// Check if the auth is actually required, as we might have
 			// updated the authmap since the datapath issued the auth
 			// required signal.
-			if i, err := a.authmap.Get(key); err == nil && i.expiration.After(time.Now()) {
+			// If the entry was cached more than authSignalBackoffTime
+			// it will authenticate again, this is to make sure that
+			// we re-authenticate if the authmap was updated by an
+			// external source.
```

Comment:
Maybe for my education, what is the harm in always authenticating? Is there a particular attack pattern or costly operation we are trying to protect the control plane against?

I'm thinking maybe if an adversarial app deploys and starts creating many many new connections immediately and it's subject to authentication, then probably in that case it is better to skip. Maybe that's the use case you have in mind?

Reply:
not sure how the one second backoff works in practice. Do you have any idea how long auth will take, probably we should add the metric for the same :thinking: ? 

IMO, with the metrics and hidden flag, user can easily tweak the value.

Reply:
Added a flag :) 
It doesn't really depend on how long auth will take but it will catch the scenario where many auth requests are created at once where it is finished but the datapath isn't up to date but the cache is. We specifically added this later as we saw this causing more unneeded handshakes and map updates. Howerver this causes manual removal of auth map entries to block traffic till the cache expired